src/test/scala/is/hail/expr/ir/IntervalSuite.scala

```diff
@@ -0,0 +1,63 @@
+package is.hail.expr.ir
+
+import org.scalatest.testng.TestNGSuite
+import org.testng.annotations.Test
+import is.hail.expr.types._
+import is.hail.TestUtils._
+import is.hail.utils.Interval
+import org.apache.spark.sql.Row
+
+class IntervalSuite extends TestNGSuite {
+
+  val tpoint1 = TTuple(TInt32())
+  val tinterval1 = TInterval(tpoint1)
+  val na = NA(tinterval1)
+
+  def point(i: Int): IR = MakeTuple(Seq(I32(i)))
+  def interval(start: IR, end: IR, includeStart: java.lang.Boolean, includeEnd: java.lang.Boolean): IR = {
+    invoke("Interval", start, end, Literal(includeStart, TBoolean()), Literal(includeEnd, TBoolean()))
+  }
+
+  val i1 = interval(point(1), point(2), true, false)
+  val i2 = interval(point(1), NA(tpoint1), true, false)
+  val i3 = interval(NA(tpoint1), point(2), true, false)
+  val i4 = interval(NA(tpoint1), point(2), null, false)
+  val i5 = interval(NA(tpoint1), point(2), true, null)
+
+  @Test def constructor() {
+    assertEvalsTo(i1, Interval(Row(1), Row(2), true, false))
+    assertEvalsTo(i2, Interval(Row(1), null, true, false))
+    assertEvalsTo(i3, Interval(null, Row(2), true, false))
+    assertEvalsTo(i4, null)
```

Comment:
So start and end are allowed to be null but not the includes start and end?

Reply:
yeah. TInterval has required booleans for includesStart and includesEnd.